OpenTok/OpenTok.cs

```diff
@@ -318,26 +584,87 @@ public Archive StartArchive(string sessionId, string name = "", bool hasVideo =
+        public async Task<Archive> StopArchiveAsync(string archiveId)
+        {
+            var request = PrepareStopArchive(archiveId);
+            string response = await Client.PostAsync(request.Url, request.Headers, request.Data);
             return JsonConvert.DeserializeObject<Archive>(response);
         }
+        #endregion
+
+        #region SetArchiveLayout
+        private OpenTokRequest PrepareSetArchiveLayout(string archiveId, ArchiveLayout layout)
+        {
+            string url = string.Format("v2/project/{0}/archive/{1}/layout", this.ApiKey, archiveId);
+            var headers = new Dictionary<string, string> { { "Content-type", "application/json" } };
+            var data = new Dictionary<string, object>();
+            if (layout != null)
+            {
+                if ((layout.Type.Equals(LayoutType.custom) && String.IsNullOrEmpty(layout.StyleSheet)) ||
+                    (!layout.Type.Equals(LayoutType.custom) && !String.IsNullOrEmpty(layout.StyleSheet)))
+                {
+                    throw new OpenTokArgumentException("Could not set the layout. Either an invalid JSON or an invalid layout options.");
+                }
+                else
+                {
+                    data.Add("type", OpenTokUtils.convertToCamelCase(layout.Type.ToString()));
+                    if (layout.Type.Equals(LayoutType.custom))
+                    {
+                        data.Add("stylesheet", layout.StyleSheet);
+                    }
+                }
+            }
+
+            return new OpenTokRequest()
+            {
+                Url = url,
+                Headers = headers,
+                Data = data
+            };
+        }
 
         /// <summary>
-        /// Returns a List of <see cref="Archive"/> objects, representing archives that are both
-        /// both completed and in-progress, for your API key.
+        /// Sets the layout type for the broadcast. For a description of layout types, see
+        /// <a href="https://tokbox.com/developer/guides/archiving/layout-control.html#dynamically-changing-the-layout-type-during-an-archive-recording">Configuring the video layout for OpenTok archive</a>.
         /// </summary>
-        /// <param name="offset">
-        /// The index offset of the first archive. 0 is offset of the most recently started archive.
-        /// 1 is the offset of the archive that started prior to the most recent archive.
-        /// </param>
-        /// <param name="count">
-        /// The number of archives to be returned. The maximum number of archives returned is 1000.
-        /// </param>
-        /// <returns>A List of <see cref="Archive"/> objects.</returns>
-        public ArchiveList ListArchives(int offset = 0 , int count = 0, string sessionId = "")
+        /// <param name="archiveId">The broadcast ID of the broadcasting session.</param>
+        /// <param name="layout">The BroadcastLayout that defines layout options for the broadcast.</param>
```

Comment:
```suggestion
        /// <param name="layout">The ArchiveLayout that defines layout options for the archive.</param>
```

(And make similar edits for the Async version of the method.